clippy_lints/src/attrs/mixed_attributes_style.rs

```diff
@@ -1,30 +1,60 @@
 use super::MIXED_ATTRIBUTES_STYLE;
 use clippy_utils::diagnostics::span_lint;
-use rustc_ast::AttrStyle;
-use rustc_lint::EarlyContext;
+use rustc_ast::{AttrKind, AttrStyle, Attribute};
+use rustc_data_structures::fx::FxHashSet;
+use rustc_lint::{LateContext, LintContext};
+use rustc_span::FileName;
 
-pub(super) fn check(cx: &EarlyContext<'_>, item: &rustc_ast::Item) {
-    let mut has_outer = false;
-    let mut has_inner = false;
+#[derive(Default)]
+struct AttrGroup {
+    inner_doc: FxHashSet<FileName>,
```

Comment:
thanks, that makes me realize I could just ignore the `attr` that are in other files when looping, instead of storing filename & find intersection...(don't know why did I think of that way first...)